io/io/src/TBufferFile.cxx

```diff
@@ -323,8 +323,13 @@ void TBufferFile::SetByteCount(ULong64_t cntpos, Bool_t packInVersion)
         && (fBufCur >= fBuffer)
         && static_cast<ULong64_t>(fBufCur - fBuffer) <= std::numeric_limits<UInt_t>::max()
         && "Byte count position is after the end of the buffer");
-   const UInt_t cnt = UInt_t(fBufCur - fBuffer) - UInt_t(cntpos) - sizeof(UInt_t);
+
    char  *buf = (char *)(fBuffer + cntpos);
+   if ((fBufCur - fBuffer - cntpos - sizeof(UInt_t)) >= kMaxMapCount) {
+      tobuf(buf, 0 | kByteCountMask);
```

Comment:
The reason to write `0 | kByteCountMask` is clear for us (today) but it might be confusing to anyone not familiar with the idea behind the plan; maybe a comment would help here.

Reply:
On the other hand, this code is already almost obsolete, i.e. replaced by https://github.com/root-project/root/pull/20574 (which is still missing a few wrinkles: decision on Reset behavior, getter and setter)